homeassistant/components/reolink/__init__.py

```diff
@@ -39,19 +40,17 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     host = ReolinkHost(hass, config_entry.data, config_entry.options)
 
     try:
-        if not await host.async_init():
-            raise ConfigEntryNotReady(
-                f"Error while trying to setup {host.api.host}:{host.api.port}: "
-                "failed to obtain data from device."
-            )
+        await host.async_init()
     except (
         ClientConnectorError,
         asyncio.TimeoutError,
         ApiError,
         InvalidContentTypeError,
+        NoDataError,
+        ReolinkException,
     ) as err:
         raise ConfigEntryNotReady(
-            f'Error while trying to setup {host.api.host}:{host.api.port}: "{str(err)}".'
+            f"Error while trying to setup {host.api.host}:{host.api.port}: '{str(err)}'"
```

Comment:
I think the extra quotes around the error just makes it harder to read the log.

Reply:
done